python/podio/pythonizations/collection_subscript.py

```diff
@@ -0,0 +1,27 @@
+"""Pythonize subscript operation for collections"""
+
+import cppyy
+from .utils.pythonizer import Pythonizer
+
+
+class CollectionSubscriptPythonizer(Pythonizer):
+    """Bound-check __getitem__ for classes derived from podio::CollectionBase"""
+
+    @classmethod
+    def priority(cls):
+        return 50
+
+    @classmethod
+    def callback(cls, class_, name):
+        def get_item(self, i):
+            if i >= len(self):
+                raise IndexError("collection index out of range")
+            return getitem_raw(self, i)
+
+        if (
+            issubclass(class_, cppyy.gbl.podio.CollectionBase)
+            and hasattr(class_, "__getitem__")
+            and hasattr(class_, "__len__")
```

Comment:
This should always be true if something is a `podio.CollectionBase` right? For `__getitem__` its the collections who implement it themselves.

Reply:
Yes
```python
            and hasattr(class_, "__getitem__")
            and hasattr(class_, "__len__")
```
are redundant. I still prefer to have them because:
- The methods are not used directly in a callback, instead the callback merely adds another method that calls them. So there will be no error that a pythonization can't be applied, the classes will just get broken.
- The exception from a  method broken with pythonization is rather cryptic: ` RecursionError: maximum recursion depth exceeded while calling a Python object`

If think it would be more readable to not apply the exception and learn from a test that it was not applied

Reply:
And yes both `__len__` and `__getitem__` are created automatically by cppyy if  `size()` and `operator()` are present